File: tests/fixtures/typescript/userRepository.ts
import { db } from './db';

export interface User {
  id: string;
  email: string;
  passwordHash: string;
}

export class UserRepository {
  async findByEmail(email: string): Promise<User | null> {
    return db.query('SELECT * FROM users WHERE email = $1', [email]);
  }

  async findById(id: string): Promise<User | null> {
    return db.query('SELECT * FROM users WHERE id = $1', [id]);
  }

  async create(email: string, passwordHash: string): Promise<User> {
    return db.query(
      'INSERT INTO users (email, password_hash) VALUES ($1, $2) RETURNING *',
      [email, passwordHash],
    );
  }
}

export default UserRepository;

File: tests/fixtures/ruby/user_repository.rb
require_relative './models/user'

# User repository fixture for coldstart parser tests.
class UserRepository
  include Enumerable

  def find_by_email(email)
    @users.find { |u| u.email == email }
  end

  def find_by_id(id)
    @users.find { |u| u.id == id }
  end

  def all
    @users
  end
end
